src/libraries/Common/src/System/Net/IPv4AddressHelper.Common.cs

```diff
@@ -105,32 +105,46 @@ internal static unsafe bool IsValid(char* name, int start, ref int end, bool all
         //                 / "2" %x30-34 DIGIT     ; 200-249
         //                 / "25" %x30-35          ; 250-255
         //
-        internal static unsafe bool IsValidCanonical(char* name, int start, ref int end, bool allowIPv6, bool notImplicitFile)
+        internal static bool IsValidCanonical<TChar>(in ReadOnlySpan<TChar> name, out int charsConsumed, bool allowIPv6, bool notImplicitFile)
```

Comment:
I'm not against changing these from pointers to spans, but was that a necessary part of this change or just opportunistic? Seems like much of the diff is because of that.

Reply:
It's mostly opportunistic, but it came from IPv6AddressHelper. I needed to adjust `IsValidStrict` and `Parse`, which accepted a pointer and a `ReadOnlySpan<char>` respectively. I couldn't use `TChar.CreateTruncating(']')` as a criteria of a `switch` block (because it's not seen as a constant by the compiler) and replacing the switch with `if` statements changed enough logic that a swap from pointers to a ROS was comparatively minor.

With that said: I think I can actually keep the switch statement in place, which probably eliminates about half of the diff.

I'll swap the spans back to pointers too though, I can see your extra comment about bounds checks.